pandas/tests/groupby/test_apply.py

```diff
@@ -1331,3 +1331,31 @@ def test_result_name_when_one_group(name):
     expected = Series([1, 2], name=name)
 
     tm.assert_series_equal(result, expected)
+
+
+def test_empty_df():
+    empty_df = DataFrame({"a": [], "b": []})
+
+    # Both operations should return an empty series instead of IndexError for apply UDF
+    result = empty_df.groupby("a", group_keys=True).b.apply(lambda x: x.values[-1])
```

Comment:
Could you compare both of these results to an actual empty Series? Just in case a code change returns the same, incorrect result for both `result` and `expected`

Reply:
@mroeschke I am having an issue equating it to an empty `series`. 
```python
    # GIVEN
    empty_series = Series([], name="b")
    empty_series.index.astype(np.float64)

    tm.assert_series_equal(result, empty_series)
```
Returns
```python
left = Float64Index([], dtype='float64', name='a'), right = RangeIndex(start=0, stop=0, step=1), obj = 'Series.index'

    def _check_types(left, right, obj="Index") -> None:
        if not exact:
            return
    
        assert_class_equal(left, right, exact=exact, obj=obj)
>       assert_attr_equal("inferred_type", left, right, obj=obj)
E       AssertionError: Series.index are different
E       
E       Attribute "inferred_type" are different
E       [left]:  floating
E       [right]: integer
```
How do I create an empty `series` that equals `result`?

Reply:
```
In [8]: ser = pd.Series([], name="b", dtype="float64", index=pd.Index([], dtype="float64", name="a"))

In [9]: ser.index
Out[9]: Float64Index([], dtype='float64', name='a')
```